CMakeLists.txt

```diff
@@ -40,5 +40,14 @@ sugar_include(.)
 option(SKIP_INSTALL "Skip the package install" OFF)
 option(XGBOOST_USE_BOOST "Use boost serialization" ON)
 option(XGBOOST_USE_HALF "Support half precision floating point storage" ON)
+option(XGBOOST_DO_LEAN "Build lean library for evaluation only" OFF)
+
+if(XGBOOST_DO_LEAN)
+  add_definitions(-DXGBOOST_DO_LEAN=1)
```

Comment:
`add_definitions` will not populate `XGBOOST_DO_LEAN` for dependent targets. If this macro used in public headers it should be `target_add_definitions(<target-name> PUBLIC XGBOOST_DO_LEAN=1)`.